### ISSUE TITLE
NamedTuple defined with Python 3.6 type hints is reported with E701

### ISSUE THREAD

**Issue by futursolo** [PyCQA/pycodestyle#635]
NamedTuple defined with Python 3.6 type hints is reported with E701 when the name of the members in the tuple starts with indent keywords.

Examples:
```python
from typing import NamedTuple, Callable


class MyNamedTuple(NamedTuple):
    format_str: str
    if_result: bool
    while_matched: Callable[..., str]
```

When testing the code above against `pycodestyle --show-source test.py`, it shows:
```
test.py:5:15: E701 multiple statements on one line (colon)
    format_str: str
              ^
test.py:6:14: E701 multiple statements on one line (colon)
    if_result: bool
             ^
test.py:7:18: E701 multiple statements on one line (colon)
    while_matched: Callable[..., str]
                 ^
```

**Comment by sigmavirus24** [comment]
Can you please provide the version of pycodestyle and the version of python it's installed on?

**Comment by futursolo** [comment]
I am running `pycodestyle 2.3.1` on `Python 3.6.0`.

**Comment by eddie-dunn** [comment]
Using the latest version of pep8 available from PyPI.

`test.py`:
```
"""Docstring"""
from typing import NamedTuple


class Foo(NamedTuple):  # pylint: disable=too-few-public-methods
    """Docstring."""
    var1: bool
    var2: str = ''
```

```
$ pep8 test.py
test.py:7:9: E701 multiple statements on one line (colon)
test.py:8:9: E701 multiple statements on one line (colon)
$ pep8 --version
1.7.0
$ python --version
Python 3.6.1
```

I get that you might not always be able to keep up with changes in Python, but this is super annoying since pep8 apparently doesn't support `# noqa` for E701 (see #180) and our CI system auto rejects code submissions if they are not PEP8 compliant.

By comparison, neither flake8 nor pylint have any problems with the code example above.

**Comment by sigmavirus24** [comment]
pep8 has been renamed to pycodestyle. Please try the latest release of pycodestyle instead of using pep8. That's why flake8 works (it uses pycodestyle).

**Comment by eddie-dunn** [comment]
Ah, fair enough, my bad. I mistakenly thought the old pep8 executable was the same version as pycodestyle, which is clearly not the case.

With that said, pycodestyle 2.3.1 works fine for my example, but not for the code @futursolo reported. Merging the two examples:

`test.py`
```
"""Docstring"""
from typing import NamedTuple


class Foo(NamedTuple):
    """Docstring."""
    format_str: str
    my_var: str
    if_result: bool
    other_var: bool
```
```
$ pycodestyle test.py
test.py:7:15: E701 multiple statements on one line (colon)
test.py:9:14: E701 multiple statements on one line (colon)
$ pycodestyle --version
2.3.1
$ python --version
Python 3.6.1
```

I.e., `format_str: str` and `if_result: bool` are wrong but `my_var: str` and ` other_var: bool` aren't? Something weird is going on.

**Comment by eddie-dunn** [comment]
I would say the problem is in the regex [on this line](https://github.com/PyCQA/pycodestyle/blob/master/pycodestyle.py#L137).

Changing it to

```python
STARTSWITH_INDENT_STATEMENT_REGEX = re.compile(
    r'^\s*({0})'.format('|'.join((s + ' ').replace(' ', r'\s+') for s in (
        'def', 'async def',
        'for', 'async for',
        'if', 'elif', 'else:',
        'try:', 'except', 'finally:',
        'with', 'async with',
        'class',
        'while',
    )))
)
```
will still make it pass the existing tests.

**Comment by eddie-dunn** [comment]
Solved by #640

**Comment by gothraven** [comment]
I'm having the same problem with

```shell
$> python --version
Python 3.7.0

$> flake8 --version
3.5.0 (mccabe: 0.6.1, pycodestyle: 2.3.1, pyflakes: 1.6.0) CPython 3.7.0 on Darwin
```

**Comment by LefterisJP** [comment]
I am also seeing this with Python 3.6.3, flake8 3.5.0 and pycodestyle being 2.3.1
https://travis-ci.org/raiden-network/raiden/jobs/414275990

**Comment by futursolo** [comment]
@LefterisJP 

This issue is resolved by a later version of pycodestyle. But flake8 pinned the version of pycodestyle to this bugged version.
Wait for the next version of flake8 and it should be fixed.

**Comment by LefterisJP** [comment]
Yeah I figured it out in the meantime. Thanks for the hint @futursolo. Added E701 to the ignored warnings for now for our project and waiting till flake8 makes another release.
